v1/format/format.go

```diff
@@ -1207,10 +1207,18 @@ func (w *writer) writeTermParens(parens bool, term *ast.Term, comments []*ast.Co
 	case ast.String:
 		if term.Location.Text[0] == '`' {
 			// To preserve raw strings, we need to output the original text,
-			// not what x.String() would give us.
 			w.write(string(term.Location.Text))
 		} else {
-			w.write(x.String())
+			// x.String() cannot be used by default because it can change the input string "\u0000" to "\x00"
+			// term.Location.Text could contain more than just the string, extract the first string it can find
+			raw := string(term.Location.Text)
+			str := regexp.MustCompile(`"[^"]*"`).FindString(raw)
```

Comment:
Regex; brave! 😝

Won't this regex fail on escaped double-ticks? E.g. `"\""`

Reply:
Can the compiled regex be stored in a global const, so we don't need to recompile it for every string?

Reply:
Nice catch I didn't consider escaped quotes, I pushed a commit dropping the use of regex. Handling escaped quotes is getting in dangerous territory making an unreadable regex haha Replaced it with `strings.Cut`, only downside is having to add the leading `"` back afterwards.